### PR TITLE
Allow Crawling Entities To Go Under Tables

### PR DESCRIPTION
**Pull request by gluesniffler**
# Description

Adds a CVAR that when enabled, allows entities to crawl under tables/flaps by lowering their DrawDepth, which in turn protects them from being targeted for projectiles while crawling. 

Additionally tables and plastic flaps were given collision properties along with reduced damage thresholds, so guns can target & destroy them easily if your mouse is on top of them. 

---

<h1>Media</h1>
<p>

https://github.com/user-attachments/assets/77a04198-11cb-4895-bf2d-6f82b7f2bb5b

</p>
</details>

---

# Changelog

:cl:
- add: Adds an optional server variable which allows entities to crawl under tables.
- tweak: Tables and plastic flaps are less resistant to damage, and can now be targeted by guns by aiming on top of them.

### CODE REVIEW

**Comment by Evgencheg** [comment]
cool! 😎

File: Content.Client/Standing/LayingDownSystem.cs [hunk 1]
```diff
@@ -48,6 +51,27 @@ private void OnMovementInput(EntityUid uid, LayingDownComponent component, MoveE
         sprite.Rotation = Angle.FromDegrees(90);
     }
 
+    private void OnDowned(DrawDownedEvent args)
+    {
+        var uid = GetEntity(args.Uid);
+
+        if (!TryComp<SpriteComponent>(uid, out var sprite) || !TryComp<LayingDownComponent>(uid, out var component))
+            return;
```
Comment (VMSolidus) [inline Content.Client/Standing/LayingDownSystem.cs:59]:
```suggestion
        if (!TryComp<SpriteComponent>(uid, out var sprite) 
            || !TryComp<LayingDownComponent>(uid, out var component))
            return;
```

File: Content.Client/Standing/LayingDownSystem.cs [hunk 2]
```diff
@@ -48,6 +51,27 @@ private void OnMovementInput(EntityUid uid, LayingDownComponent component, MoveE
...
+        if (!component.OriginalDrawDepth.HasValue)
+            component.OriginalDrawDepth = sprite.DrawDepth;
+
+        sprite.DrawDepth = (int) DrawDepth.SmallMobs;
+    }
+
+    private void OnStood(EntityUid uid, LayingDownComponent component, StoodEvent args)
+    {
+        if (!TryComp<SpriteComponent>(uid, out var sprite) || !component.OriginalDrawDepth.HasValue)
+            return;
```
Comment (VMSolidus) [inline Content.Client/Standing/LayingDownSystem.cs:70]:
```suggestion
        if (!TryComp<SpriteComponent>(uid, out var sprite) 
            || !component.OriginalDrawDepth.HasValue)
            return;
```

**Review by VMSolidus — APPROVED** [review]
We can add the bullet click interactions to far more objects as we go!

File: Content.Client/Standing/LayingDownSystem.cs [hunk 3]
```diff
@@ -48,6 +51,29 @@ private void OnMovementInput(EntityUid uid, LayingDownComponent component, MoveE
...
+        var uid = GetEntity(args.Uid);
+
+        if (!TryComp<SpriteComponent>(uid, out var sprite) 
+            || !TryComp<LayingDownComponent>(uid, out var component))
+            return;
+
+        if (!component.OriginalDrawDepth.HasValue)
+            component.OriginalDrawDepth = sprite.DrawDepth;
+
+        sprite.DrawDepth = (int) DrawDepth.SmallMobs;
```
Comment (FoxxoTrystan) [inline Content.Client/Standing/LayingDownSystem.cs:65]: Unable to make a proper test but, make sure that we cannot crawl under tables...
Wait no this change sprite, not fixtures.

Carry on!

**Review by FoxxoTrystan — APPROVED** [review]

**Pull request merged**